src/fidesops/core/config.py

```diff
@@ -59,6 +59,26 @@ class RedisSettings(FidesSettings):
     ENABLED: bool = True
     SSL: bool = False
     SSL_CERT_REQS: Optional[str] = "required"
+    REDIS_CONNECTION_URL: Optional[str] = None
+
+    @validator("REDIS_CONNECTION_URL", pre=True)
+    @classmethod
+    def assemble_redis_connection_url(
+        cls,
+        v: Optional[str],
+        values: Dict[str, str],
+    ) -> str:
+        """Join Redis connection credentials into a connection string"""
+        if isinstance(v, str):
+            # If the whole URL is provided via the config, preference that
+            return v
+
+        return "redis://:{password}@{host}:{port}/{db_index}".format(
+            password=values["PASSWORD"],
+            host=values["HOST"],
+            port=values["PORT"],
+            db_index=values.get("DB_INDEX"),
+        )
```

Comment:
f-strings are slightly faster. Also `DB_INDEX` is a required field so you don't need the `.get` since you know it will have a value.

```suggestion
        return f"redis://:{values['PASSWORD']}@{values['HOST']}:{values['PORT']}/{values['DB_INDEX']}"
```

Reply:
Have opted to make `DB_INDEX` optional, as the connection string will still work without it